common/thorhelper/thorsoapcall.cpp

```diff
@@ -2544,23 +2579,30 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                 bool keepAlive2;
                 StringBuffer contentType;
                 int rval = readHttpResponse(response, socket, keepAlive2, contentType);
+                socketOperationSpan->setSpanAttribute("http.response.status_code", (int64_t)rval);
                 keepAlive = keepAlive && keepAlive2;
 
                 if (soapTraceLevel > 4)
                     master->logctx.CTXLOG("%s: received response (%s) from %s:%d", getWsCallTypeName(master->wscType),master->service.str(), url.host.str(), url.port);
 
                 if (rval != 200)
                 {
+                    socketOperationSpan->setSpanStatus(false);
```

Comment:
I see you have inverted the condition.  It would be clearer if the function was renamed to

setSpanSuccess(true|false)

 or something that makes it clear what the parameter means.
(Also now not needed because of the recordError() calls).

Reply:
changing, but this is def still needed regardless of the recordError calls.